src/components/app.js

```diff
@@ -87,10 +89,12 @@ class App extends Component {
                         <PrivateRoute exact path="/" user={this.state.user} component={(props) => <Welcome {...props} user={this.state.user} />} />
                         <Route path="/signin" component={(props) => <SignIn user={this.state.user} editUser={this.editUser} />} />
                         <Route path="/signup" component={(props) => <SignUp user={this.state.user} editUser={this.editUser} />} />
+                        <Route path="/user/:userID" component={(props) => <UpdateUser user={this.state.user} editUser={this.editUser} />} />
                         <Route path="/about" component={About} />
                         <Route path="/event" component={(props) => <Event user={this.state.user} />} />
                         <Route path="/eventlist" component={(props) => <EventList user={this.state.user} />} />
                         <Route path="/createevent" component={(props) => <CreateEvent user={this.state.user} />} />
+                        <Route path="/joinevent/:eventID" component={(props) => <UpdateEvent user={this.state.user} />} />
```

Comment:
For UpdateUser and UpdateEvent, you should probably not pass userID and eventID in the url. Could you try to do it the way it works for eventinfo? You can see in lines 30-47 of eventlist.js how to pass parameters with a NavLink, and you can see examples of how to access these parameters that you pass in 52, 58, and 61 of eventinfo.js.